modules/repository/repo.go

```diff
@@ -240,14 +237,14 @@ func MigrateRepositoryGitData(ctx context.Context, u *user_model.User,
 
 // cleanUpMigrateGitConfig removes mirror info which prevents "push --all".
 // This also removes possible user credentials.
-func cleanUpMigrateGitConfig(configPath string) error {
-	cfg, err := ini.Load(configPath) // FIXME: the ini package doesn't really work with git config files
-	if err != nil {
-		return fmt.Errorf("open config file: %w", err)
-	}
-	cfg.DeleteSection("remote \"origin\"")
-	if err = cfg.SaveToIndent(configPath, "\t"); err != nil {
-		return fmt.Errorf("save config file: %w", err)
+func cleanUpMigrateGitConfig(ctx context.Context, repoPath string) error {
+	cmd := git.NewCommand(ctx, "remote", "rm", "origin")
+	// if the origin does not exist
+	_, _, err := cmd.RunStdString(&git.RunOpts{
+		Dir: repoPath,
+	})
+	if err != nil && !strings.HasPrefix(err.Error(), "fatal: No such remote") {
```

Comment:
It should be `!strings.Contains`, or use `err.Stderr()`

Because the returned `err.Error()` is generated by `ConcatenateError`, which outputs `exit status ... - fatal: ...`, that's a very old design in Gitea's code base.

Reply:
Done.